assembly/src/parsers/mod.rs

```diff
@@ -85,44 +81,53 @@ pub struct ModuleAst {
 
 impl ModuleAst {
     /// Returns byte representation of the `ModuleAst`.
-    pub fn to_bytes(&self) -> Result<Vec<u8>, SerializationError> {
-        let mut byte_writer = ByteWriter::default();
-
-        // docs
-        self.docs.write_into(&mut byte_writer)?;
-
-        // local procedures
-        self.local_procs.write_into(&mut byte_writer)?;
-
-        Ok(byte_writer.into_bytes())
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut target = Vec::<u8>::default();
+        self.write_into(&mut target);
+        target
     }
 
     /// Returns a `ModuleAst` struct by its byte representation.
-    pub fn from_bytes(bytes: &[u8]) -> Result<Self, SerializationError> {
-        let mut byte_reader = ByteReader::new(bytes);
-
-        // docs
-        let docs = Deserializable::read_from(&mut byte_reader)?;
-
-        // local procedures
-        let local_procs = Deserializable::read_from(&mut byte_reader)?;
-
-        Ok(ModuleAst { docs, local_procs })
+    pub fn from_bytes(bytes: &[u8]) -> Result<Self, DeserializationError> {
+        let mut source = SliceReader::new(bytes);
+        Self::read_from(&mut source)
     }
 }
 
 impl Serializable for ModuleAst {
-    fn write_into(&self, target: &mut ByteWriter) -> Result<(), SerializationError> {
-        self.docs.write_into(target)?;
-        self.local_procs.write_into(target)?;
-        Ok(())
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        match &self.docs {
+            Some(t) => {
+                target.write_bool(true);
+                target.write_u64(t.len() as u64);
+                target.write_bytes(t.as_bytes());
+            }
+            None => {
+                target.write_bool(false);
+            }
+        }
+        target.write_u64(self.local_procs.len() as u64);
```

Comment:
I think `u64` both for docs and for the number of procedures is too much - let's use `u16` and add TODOs to enforce these restrictions via the constructor.